OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -129,24 +136,24 @@ void INotifyKilled.Killed(Actor self, AttackInfo e)
 			weapon.Impact(Target.FromPos(self.CenterPosition), source);
 		}
 
-		WeaponInfo ChooseWeaponForExplosion(Actor self)
+		WeaponInfo ChooseWeaponForExplosion(Actor self, int sharedRandom)
 		{
-			var armaments = self.TraitsImplementing<Armament>();
-			if (!armaments.Any())
+			if (!anyArmaments)
 				return Info.WeaponInfo;
+			else if (sharedRandom > Info.LoadedChance)
+				return Info.EmptyWeaponInfo;
+
+			// PERF: Avoid LINQ
+			foreach (var a in armaments)
+				if (!a.IsReloading)
+					return Info.WeaponInfo;
 
-			// TODO: EmptyWeapon should be removed in favour of conditions
```

Comment:
Is that `TODO` really resolved?

Reply:
Huh, I thought I commented on this already, seems I forgot...

Anyway, previous attempts have shown that to be rather complicated and requiring some serious thought put into any possible solution, to a degree that I honestly believe it's just not worth the hassle when the current solution works well enough.
Considering just how many things we have on our plate that are more pressing or worthwhile, I just don't feel this one is worth putting any more effort into.

That's ultimately just my stance, if you'd prefer to keep the comment anyway I guess I can restore it.